include/wil/win32_helpers.h

```diff
@@ -18,6 +18,21 @@
 #include <PathCch.h>
 #include <objbase.h>
 
+// detect std::bit_cast
+// should replace with __WI_LIBCPP_CONSTEXPR_AFTER_CXX20 once compilers
+// correctly signal C++20 in __cplusplus
+#ifdef __has_include
+#  if __has_include(<version>)
+#    include <version>
+#    ifdef __cpp_lib_bit_cast
+#      include <bit>
+#      define WI_CONSTEXPR_BIT_CAST __WI_LIBCPP_CONSTEXPR

```

Comment:
In general, names with the `WI_` prefix are intended to be consumable outside of WIL. The prefix `__WI_` - disregarding the fact that this violates the standard, we know - is typically used for internal macros.

That said, since the the implementation of the function dramatically changes based off feature presence, this definition doesn't really seem necessary. See later comment

Reply:
There's also the issue that this macro goes undefined if `__has_include` is not defined or `<version>` is unavailable.